tests/components/proximity/test_init.py

```diff
@@ -1,557 +1,589 @@
 """The tests for the Proximity component."""
-import unittest
 
-from homeassistant.components import proximity
 from homeassistant.components.proximity import DOMAIN
-from homeassistant.setup import setup_component
-
-from tests.common import get_test_home_assistant
-
-
-class TestProximity(unittest.TestCase):
-    """Test the Proximity component."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.hass.config.components.add("zone")
-        self.hass.states.set(
-            "zone.home",
-            "zoning",
-            {"name": "home", "latitude": 2.1, "longitude": 1.1, "radius": 10},
-        )
-        self.hass.states.set(
-            "zone.work",
-            "zoning",
-            {"name": "work", "latitude": 2.3, "longitude": 1.3, "radius": 10},
-        )
-        self.addCleanup(self.hass.stop)
-
-    def test_proximities(self):
-        """Test a list of proximities."""
-        config = {
-            "proximity": {
-                "home": {
-                    "ignored_zones": ["work"],
-                    "devices": ["device_tracker.test1", "device_tracker.test2"],
-                    "tolerance": "1",
-                },
-                "work": {"devices": ["device_tracker.test1"], "tolerance": "1"},
-            }
+from homeassistant.setup import async_setup_component
+
+
+async def test_proximities(hass):
+    """Test a list of proximities."""
+    config = {
+        "proximity": {
+            "home": {
+                "ignored_zones": ["work"],
+                "devices": ["device_tracker.test1", "device_tracker.test2"],
+                "tolerance": "1",
+            },
+            "work": {"devices": ["device_tracker.test1"], "tolerance": "1"},
         }
+    }
 
-        assert setup_component(self.hass, DOMAIN, config)
+    assert await async_setup_component(hass, DOMAIN, config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)